app/helpers/compile_arm_gnu.rb

```diff
@@ -0,0 +1,30 @@
+module CompileArmGnu
+  # Compilation module for the arm-eabi-non-gcc toolchain
+  
+  module_function
+  def run(submission, tests)
+    comments = ""
+    hash = Digest::SHA1.hexdigest("#{rand(10000)}#{Time.now}")
+
+    # Write the submission
+    folder = "#{Rails.root}/tmp/compilations"
+    libraries = "#{Rails.root}/Library/ARM GNU"
+
+    `mkdir -p #{folder}`
+    submission_contents = submission.plaintext.gsub("\r\n","\n").gsub("\r","\n")
```

Comment:
Currently only plaintext assignment types are supported for compilation, i.e one file. If we need to add zip compilation support, then that is a new feature.